src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -184,7 +184,14 @@ void OTARequestor::OnQueryImageFailure(void * context, CHIP_ERROR error)
     OTARequestor * requestorCore = static_cast<OTARequestor *>(context);
     VerifyOrDie(requestorCore != nullptr);
 
-    ChipLogDetail(SoftwareUpdate, "QueryImage failure response %" CHIP_ERROR_FORMAT, error.Format());
+    ChipLogError(SoftwareUpdate, "Received QueryImage failure response: %" CHIP_ERROR_FORMAT, error.Format());
+
+    if (error == CHIP_ERROR_TIMEOUT)
```

Comment:
Would we have to cleanup the CASE session in case of any other errors?

Reply:
If the errors are not related to CASE, then probably no. This tear down and reestablishment is expensive so would like to avoid unless absolutely necessary.